### PR TITLE
[CMR-71] SO upload limited

### PR DESCRIPTION
**Pull request by GabrielMoreschi**
Foi criada uma função que identifica o sistema operacional do usuário e limita visualmente as opções de arquivos para utilizar na ferramenta de upload.

### CODE REVIEW

File: components/map/file-loader/FileLoaderControl.vue [hunk 1]
```diff
@@ -220,6 +220,7 @@ export default {
     data() {
         return {
             showOptions: false,
+            name: '',
```
[the remark this answers is not in the document]
Comment (GabrielMoreschi) [reply components/map/file-loader/FileLoaderControl.vue]: Resolved

**Review by dagnaldo — CHANGES_REQUESTED** [review]
Check suggestions from daily meeting.

**Comment by GabrielMoreschi** [comment]
> Check suggestions from daily meeting.

Resolved

File: components/map/file-loader/FileLoaderControl.vue [hunk 2]
```diff
@@ -411,21 +415,24 @@ export default {
...
-            this.readFile(f, reader)
+            const loadMethod = this.loadFileMethod(type)
+            reader.addEventListener('load', (evt) =>
+                loadMethod(evt.target, f.name)
+            )
+            this.readFile(f, reader, type)
         },
         ...mapMutations('map', ['addFileToMap', 'removeFileFromMap']),
-        ...mapActions('map', ['changeStyle']),
+        ...mapActions('map', ['changeStyle', 'saveToDatabase']),
```
Comment (dagnaldo) [inline components/map/file-loader/FileLoaderControl.vue:435]: Save to database is working?

File: components/map/file-loader/FileLoaderControl.vue [hunk 3]
```diff
@@ -331,35 +325,34 @@ export default {
...
             else this.fileError(null, this.$i18n.t('file-error-parsing-kmlz'))
         },
-
+        loadGpxFile(evt, name) {
+            const geoJson = omnivore.gpx.parse(evt.result).toGeoJSON()
+            if (geoJson.features && geoJson.features.length)
+                this.addToMap(geoJson, name)
+            else this.fileError(null, this.$i18n.t('file-error-parsing-gpx'))
+        },
         loadKmzFile(evt, name) {
```
[the remark this answers is not in the document]
Comment (GabrielMoreschi) [reply components/map/file-loader/FileLoaderControl.vue:353]: Resolved

File: components/map/file-loader/FileLoaderControl.vue [hunk 4]
```diff
@@ -271,54 +272,47 @@ export default {
             this.files.splice(index, 1)
             this.removeFileFromMap(index)
         },
-
+        saveIntoDb(index) {
```
[the remark this answers is not in the document]
Comment (GabrielMoreschi) [reply components/map/file-loader/FileLoaderControl.vue:275]: Resolved

File: components/map/file-loader/FileLoaderControl.vue [hunk 5]
```diff
@@ -74,6 +74,11 @@
                                     </template>
                                     <span>{{ $t('go-to-tooltip') }}</span>
                                 </v-tooltip>
+                                <ConfirmButton
```
[the remark this answers is not in the document]
Comment (GabrielMoreschi) [reply components/map/file-loader/FileLoaderControl.vue:77]: Resolved

File: components/map/file-loader/FileLoaderControl.vue [hunk 6]
```diff
@@ -331,35 +325,34 @@ export default {
...
                 })
                 .catch((error) =>
                     this.fileError(error, this.$i18n.t('file-error-shapefile'))
                 )
         },
-
         loadKmlFile(evt, name) {
-            const geoJson = omnivore.kml.parse(evt.target.result).toGeoJSON()
-
+            const geoJson = omnivore.kml.parse(evt.result).toGeoJSON()
```
[the remark this answers is not in the document]
Comment (GabrielMoreschi) [reply components/map/file-loader/FileLoaderControl.vue:342]: yes

File: components/map/file-loader/FileLoaderControl.vue (hunk 2 above)
Comment (GabrielMoreschi) [reply components/map/file-loader/FileLoaderControl.vue:435]: removed

File: components/map/file-loader/FileLoaderControl.vue [hunk 7]
```diff
@@ -254,24 +258,30 @@ export default {
...
                 ) || false
             )
         },
+
         remove(index) {
             this.files.splice(index, 1)
             this.removeFileFromMap(index)
         },
 
+        saveIntoDb(index) {
```
Comment (dagnaldo) [inline components/map/file-loader/FileLoaderControl.vue:281]: Is it in use?

**Review by dagnaldo — CHANGES_REQUESTED** [review]

**Review by dagnaldo — APPROVED** [review]

**Pull request merged**